static/app/components/smartSearchBar/index.tsx

```diff
@@ -350,45 +278,61 @@ class SmartSearchBar extends React.Component<Props, State> {
   }
 
   componentWillUnmount() {
+    this.inputResizeObserver?.disconnect();
+
     if (this.blurTimeout) {
       clearTimeout(this.blurTimeout);
-      this.blurTimeout = undefined;
     }
   }
 
   /**
    * Tracks the dropdown blur
    */
   blurTimeout?: number;
+
   /**
    * Ref to the search element itself
    */
   searchInput = React.createRef<HTMLTextAreaElement>();
 
-  blur = () => {
-    if (!this.searchInput.current) {
-      return;
-    }
-    this.searchInput.current.blur();
-  };
+  /**
+   * Ref to the search container
+   */
+  containerRef = React.createRef<HTMLDivElement>();
 
-  onSubmit = (evt: React.FormEvent) => {
-    const {organization, savedSearchType} = this.props;
-    evt.preventDefault();
+  /**
+   * Used to determine when actions should be moved to the action overflow menu
+   */
+  inputResizeObserver: ResizeObserver | null = null;
 
-    trackAnalyticsEvent({
-      eventKey: 'search.searched',
-      eventName: 'Search: Performed search',
-      organization_id: organization.id,
-      query: removeSpace(this.state.query),
-      search_type: savedSearchType === 0 ? 'issues' : 'events',
-      search_source: 'main_search',
-    });
+  /**
+   * Updates the actionsVisible count as the search bar is resized
+   */
+  updateActionsVisible = (entries: ResizeObserverEntry[]) => {
+    const entry = entries[0];
```

Comment:
Is `entries.length` always > 0?

Reply:
Yeah since we're only observing one thing. I can add a undefined check here just to be safe though, that's a good idea.